filesystem/Upload.php

```diff
@@ -335,6 +344,11 @@ class Upload_Validator {
 	 */
 	public $allowedExtensions = array();
 
+	public function __construct() {
+		$fileSize = Config::inst()->get('Upload_Validator', 'default_max_file_size');
+		$this->setAllowedMaxFileSize($fileSize);
```

Comment:
This function does a fair bit of work, which may be unnecessary if the defaults are overridden after construct then this is a waste of CPU cycles.

Could there be a way to leave the computation to be "just in time" (ie: only when it's manually set OR when the `getAllowedMaxFileSize` is executed.